test/quantization/core/test_workflow_ops.py

```diff
@@ -693,15 +693,19 @@ def test_forward_per_channel(self, device, X):
     def _test_forward_per_channel_cachemask_impl(self, device):
         torch_types = (torch.qint8, torch.quint8)
         float_types = (torch.float32, torch.float16, torch.float64)
-        for torch_type, float_type in itertools.product(torch_types, float_types):
+        if device == 'cpu':
+            zero_point_types = (torch.int, torch.float32, torch.float16)
+        else:
+            zero_point_types = (torch.int,)
+        for torch_type, float_type, zero_point_type in itertools.product(torch_types, float_types, zero_point_types):
             X = torch.randn(1, 2, 4, 4, dtype=float_type).to(device)
             # pick the scale + zp so that some values get clipped
             axis = 1
             obs = torch.quantization.PerChannelMinMaxObserver(axis, torch_type).to(device)
             obs(X * 0.75)
             scale, zero_point = obs.calculate_qparams()
             # TODO(future PR): fix the wrong dtype in obs.calculate_qparams and remove the cast
-            zero_point = zero_point.to(torch.int32)
+            zero_point = zero_point.to(zero_point_type)
```

Comment:
For any of these tests, do we need to also update the reference implementation to have the same numerics as the op for floating point zero point?

Reply:
The reference implementation should already match the numerics for floating point zero_point. In the reference impl, at `common_quantized.py:185`, rounding happens like `round(X * 1/scale + zero_point)`, similar to this floating point zero_point implementation in the op.

Somewhat related, I would have expected the reference implementation to match the existing integer op implementation of `static_cast<int64_t>(zero_point + std::nearbyint(self * inv_scale))`.  This might be a good follow-up issue to make sure the reference implementation matches the numerics for existing int zero_point op implementation.

Reply:
Thanks, would be great if we can track the follow-up as a TODO in the test.